app/helpers/samples_helper.rb

```diff
@@ -100,38 +99,55 @@ def samples_output_data(samples)
     final_result
   end
 
+  def filter_samples(samples, query)
+    samples = if query == 'WAITING'
+                samples.joins("LEFT OUTER JOIN pipeline_runs ON pipeline_runs.sample_id = samples.id").where("samples.status = ?  or pipeline_runs.job_status is NULL and pipeline_runs.finalized != 1", 'created')
+              elsif query == 'FAILED'
+                samples.joins("INNER JOIN pipeline_runs ON pipeline_runs.sample_id = samples.id").where(status: 'checked').where("pipeline_runs.id in (select max(id) from pipeline_runs group by sample_id)").where("pipeline_runs.job_status IN (?) and pipeline_runs.finalized = 1", [query || 'ERROR'])
```

Comment:
this is problematic. "ERROR" is also a transient status and the job status will say: "step-name FAILED" 
It should be changed to 
samples.joins("INNER JOIN pipeline_runs ON pipeline_runs.sample_id = samples.id").where(status: 'checked').where("pipeline_runs.id in (select max(id) from pipeline_runs group by sample_id)").where("pipeline_runs.job_status like '%FAILED' ")
and in this case you don't have to change finalized at all.